### PR TITLE
Kit 166

### PR DESCRIPTION
**Pull request by emmirose**
Added delele a task, delete a todo, create a todo, add to favorite

### CODE REVIEW

**Review by Newaluigi — APPROVED** [review]

File: src/app/components/todolist/todolist.component.html [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
+          </button>
+          <mat-menu #menu="matMenu">
+            <button mat-menu-item>
+              <i class="fa-solid fa-pencil" title="Modifier"></i>
+              Modifier
+            </button>
+            <!-- <button mat-menu-item (click)="deleteTask(task)"> -->
+            <i class="fa-solid fa-trash" title="Supprimer"></i>
+            Supprimer
+            <!-- </button> -->
```
Comment (Benja-san) [inline src/app/components/todolist/todolist.component.html:45]: remove dead code

File: src/app/pages/todolists/todolists.component.ts [hunk 2]
```diff
@@ -1,20 +1,78 @@
...
+  constructor(public dialogRef: MatDialogRef<NewListFormComponent>, private todoService: TodoService) {}
+  todolists: ToDoList[] = [];
+  todolist!: ToDoList;
+  title = new FormControl('');
+  description = new FormControl('');
+  tasks = new FormControl('');
+
+  createTodolist(): void {
+    const title = this.title.value as string;
+    const description = this.description.value as string;
```
Comment (Benja-san) [inline src/app/pages/todolists/todolists.component.ts:65]: does it work if you don't put any description ?

File: src/app/pages/todolists/todolists.component.ts [hunk 3]
```diff
@@ -1,20 +1,78 @@
...
+
+    const body = {
+      title: title,
+      description: description,
+    };
+
+    this.todoService.createList(body).subscribe((data) => {
+      this.todolist = data;
+      this.todolist = new ToDoList(this.todolist.id, title as string, [], description as string, false);
```
Comment (Benja-san) [inline src/app/pages/todolists/todolists.component.ts:74]: I'm wondering what's the id of the new todo ?

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/components/todolist/todolist.component.html (hunk 1 above)
Comment (emmirose) [reply src/app/components/todolist/todolist.component.html:45]: All good, I needed this button

File: src/app/pages/todolists/todolists.component.ts (hunk 2 above)
Comment (emmirose) [reply src/app/pages/todolists/todolists.component.ts:65]: Yes it does !

File: src/app/pages/todolists/todolists.component.ts (hunk 3 above)
Comment (emmirose) [reply src/app/pages/todolists/todolists.component.ts:74]: It's the id generated in the DB !

Comment (Benja-san) [reply src/app/pages/todolists/todolists.component.ts:74]: all good !

**Review by Newaluigi — APPROVED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**